config/network/gossipsub_rpc_inspectors.go

```diff
@@ -0,0 +1,140 @@
+func (conf *GossipSubRPCValidationInspectorConfigs) GetCtrlMsgValidationConfig(controlMsg p2p.ControlMessageType) (*CtrlMsgValidationConfig, bool) {
+	switch controlMsg {
+	case p2p.CtrlMsgGraft:
+		return conf.GraftLimits, true
+	case p2p.CtrlMsgPrune:
+		return conf.PruneLimits, true
+	case p2p.CtrlMsgIHave:
+		return conf.IHaveLimits, true
+	default:
+		return nil, false
+	}
+}
+
+// AllCtrlMsgValidationConfig returns all control message validation configs in a list.
+func (conf *GossipSubRPCValidationInspectorConfigs) AllCtrlMsgValidationConfig() CtrlMsgValidationConfigs {
+	return CtrlMsgValidationConfigs{conf.GraftLimits, conf.PruneLimits, conf.IHaveLimits}
+}
+
+// CtrlMsgValidationConfigs list of *CtrlMsgValidationConfig
+type CtrlMsgValidationConfigs []*CtrlMsgValidationConfig
+
+// CtrlMsgValidationConfig configuration values for upper, lower threshold and rate limit.
+type CtrlMsgValidationConfig struct {
+	// ControlMsg the type of RPC control message.
+	ControlMsg p2p.ControlMessageType `mapstructure:"control-message-type"`
+	// HardThreshold specifies the hard limit for the size of an RPC control message.
+	// While it is generally expected that RPC messages with a size greater than HardThreshold should be dropped,
+	// there are exceptions. For instance, if the message is an 'iHave', blocking processing is performed
+	// on a sample of the control message rather than dropping it.
+	HardThreshold uint64 `mapstructure:"hard-threshold"`
+	// SafetyThreshold specifies the lower limit for the size of the RPC control message, it is safe to skip validation for any RPC messages
+	// with a size < SafetyThreshold. These messages will be processed as soon as possible.
+	SafetyThreshold uint64 `mapstructure:"safety-threshold"`
+	// RateLimit number of allowed messages per second, use 0 to disable rate limiting.
+	RateLimit int `mapstructure:"rate-limit"`
+	// rateLimiter basic limiter without lockout duration.
+	rateLimiter p2p.BasicRateLimiter
+}
+
+// Validate validates control message validation limit values.
+func (c *CtrlMsgValidationConfig) Validate() error {
+	// check common config values used by all control message types
+	switch {
+	case c.RateLimit < 0:
+		return NewInvalidLimitConfigErr(c.ControlMsg, fmt.Errorf("invalid rate limit value %d must be greater than 0", c.RateLimit))
+	case c.HardThreshold <= 0:
+		return NewInvalidLimitConfigErr(c.ControlMsg, fmt.Errorf("invalid hard threshold value %d must be greater than 0", c.HardThreshold))
+	case c.SafetyThreshold <= 0:
+		return NewInvalidLimitConfigErr(c.ControlMsg, fmt.Errorf("invalid safety threshold value %d must be greater than 0", c.SafetyThreshold))
+	}
+	return nil
+}
+
+func (c *CtrlMsgValidationConfig) SetRateLimiter(r p2p.BasicRateLimiter) {
+	c.rateLimiter = r
+}
+
+func (c *CtrlMsgValidationConfig) RateLimiter() p2p.BasicRateLimiter {
+	return c.rateLimiter
+}
```

Comment:
Might make more sense to put this into `ControlMsgValidationInspector`, since it is essentially a dependency of that component. Feels a bit out of place here, since it is created in the constructor, rather than read in from flags/config file. Not a big deal though.

Reply:
https://github.com/onflow/flow-go/pull/4340/commits/b35ae57f69e529f440036312f964143b3a7cef8b